### PR TITLE
Non cartesian traj

### PR DESCRIPTION
**Pull request by chaithyagr**
Finally have a test and fixed codes in earlier case.

### CODE REVIEW

File: tf_fastmri_data/datasets/non_cartesian.py [hunk 1]
```diff
@@ -93,14 +93,14 @@ def preprocessing(self, image, kspace):
...
         nc_kspace = nufft(self.nufft_obj, orig_image_channels, traj, self.image_size, multicoil=self.multicoil)
         nc_kspace, image = scale_tensors(nc_kspace, image, scale_factor=self.scale_factor)
         image = image[..., None]
         nc_kspaces_channeled = nc_kspace[..., None]
         orig_shape = tf.ones([tf.shape(kspace)[0]], dtype=tf.int32) * self.image_size[-1]
-        if self.crop_image_data:
-            image = adjust_image_size(image, self.image_size)
-        else:
+        if not self.crop_image_data:
             output_shape = tf.shape(image)[1:][None, :]
```
Comment (zaccharieramzi) [inline tf_fastmri_data/datasets/non_cartesian.py:104]: Won't this be a problem now because `tf.shape(image)[1:]` is going to be a tensor of length 3?

I mean not a problem per se, because we slice anyway in the network, but it carries extra unnecessary info.
Maybe you could do this `if` a bit earlier (like as the `else` of the modification above)

File: tf_fastmri_data/tests/datasets/non_cartesian_test.py [hunk 2]
```diff
@@ -5,12 +5,12 @@
 
 
 
-@pytest.mark.parametrize('acq_type', ['radial', 'cartesian', 'spiral'])
-@pytest.mark.parametrize('dcomp', [True, False])
+@pytest.mark.parametrize('acq_type', ['radial'])
+@pytest.mark.parametrize('dcomp', [True])
 @pytest.mark.parametrize('multicoil', [True, False])
-@pytest.mark.parametrize('slice_random', [True, False])
+@pytest.mark.parametrize('slice_random', [False, True])
```
Comment (zaccharieramzi) [inline tf_fastmri_data/tests/datasets/non_cartesian_test.py:11]:
```suggestion
@pytest.mark.parametrize('slice_random', [True, False])
```

Unnecessary change I think, let's keep it as is

File: tf_fastmri_data/preprocessing_utils/extract_smaps.py [hunk 3]
```diff
@@ -55,9 +55,9 @@ def non_cartesian_extract_smaps(kspace, trajs, dcomp, nufft_back, shape, low_fre
     # Obtain the trajectory, kspace and density compensation for low frequency
     low_freq_traj = tf.boolean_mask(trajs, low_freq_bool_mask, axis=2)
     low_freq_kspace = tf.boolean_mask(kspace, low_freq_bool_mask, axis=2)
-    low_freq_dcomp = tf.boolean_mask(dcomp, low_freq_bool_mask, axis=1)
+    low_freq_dcomp = tf.boolean_mask(dcomp[:, None, ...], low_freq_bool_mask, axis=2)
```
Comment (zaccharieramzi) [inline tf_fastmri_data/preprocessing_utils/extract_smaps.py:58]:
```suggestion
    low_freq_dcomp = tf.boolean_mask(dcomp[:, None], low_freq_bool_mask, axis=2)
```

File: tf_fastmri_data/preprocessing_utils/extract_smaps.py [hunk 4]
```diff
@@ -55,9 +55,9 @@ def non_cartesian_extract_smaps(kspace, trajs, dcomp, nufft_back, shape, low_fre
...
     coil_smap = nufft_back(low_freq_kspace * tf.cast(low_freq_dcomp, kspace.dtype), low_freq_traj)
     coil_smap = adjust_image_size(coil_smap, shape, multicoil=True)
-    low_freq_rss = tf.norm(coil_smap, axis=1)
+    low_freq_rss = tf.norm(coil_smap, axis=1)[:, None, ...]
```
Comment (zaccharieramzi) [inline tf_fastmri_data/preprocessing_utils/extract_smaps.py:61]:
```suggestion
    low_freq_rss = tf.norm(coil_smap, axis=1)[:, None]
```

**Review by zaccharieramzi — CHANGES_REQUESTED** [review]
A few cosmetics

File: tf_fastmri_data/datasets/non_cartesian.py (hunk 1 above)
Comment (chaithyagr) [reply tf_fastmri_data/datasets/non_cartesian.py:104]: I dont think it will.. I had to do this mainly cause the line `image = image[..., None]` would make the size bigger than what is needed by input for `adjust_image_size` function. Rest of the code is pretty much the same.

Comment (zaccharieramzi) [reply tf_fastmri_data/datasets/non_cartesian.py:104]: honestly I am fine with it nv mind

**Review by zaccharieramzi — APPROVED** [review]
LGTM

**Comment by chaithyagr** [comment]
Yaay, 5min for test, and we have 8 test cases for non cartesian. I guess this is  good!

Can we get rid of this travis mess? 

Just letting you know that this is ready for merge now.
